packages/ui/src/2_molecules/AmountInput/AmountInput.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+  ReactNode,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import classNames from 'classnames';
+
+import { InputProps } from '../../1_atoms';
+import { InputBase } from '../../1_atoms/InputBase/InputBase';
+import { HelperButton } from '../HelperButton/HelperButton';
+import styles from './AmountInput.module.css';
+
+const DEFAULT_DECIMAL_PRECISION = 6;
+
+export enum AmountInputVariant {
+  large = 'large',
+  small = 'small',
+}
+
+type AmountInputProps = Omit<InputProps, 'classNameInput' | 'type' | 'size'> & {
+  label?: ReactNode;
+  variant?: AmountInputVariant;
+  tooltip?: ReactNode;
+  useAmountButtons?: boolean;
+  decimalPrecision?: number;
+  unit?: ReactNode;
+  maxAmount?: number;
+};
+
+export const AmountInput = React.forwardRef<HTMLInputElement, AmountInputProps>(
+  (
+    {
+      className,
+      tooltip,
+      variant = AmountInputVariant.large,
+      useAmountButtons = false,
+      decimalPrecision = DEFAULT_DECIMAL_PRECISION,
+      unit,
+      maxAmount,
+      label,
+      invalid,
+      dataLayoutId,
+      value,
+      ...rest
+    },
+    ref,
+  ) => {
+    const inputRef = useRef<HTMLInputElement>(
+      null,
+    ) as React.MutableRefObject<HTMLInputElement>;
+
+    useImperativeHandle(ref, () => inputRef.current);
+
+    const [focused, setFocused] = useState(false);
+    const onFocus = useCallback(() => setFocused(true), []);
+    const onBlur = useCallback(() => setFocused(false), []);
+
+    const formattedValue = useMemo(() => {
+      if (!value) {
+        return 0;
+      }
+      if (!decimalPrecision) {
+        return value;
+      }
+
+      const decimalLength = value.toString().split(/[,.]/)[1]?.length || 0;
+      if (decimalLength <= decimalPrecision) {
+        return value;
+      }
+
+      const unformattedValue =
+        typeof value === 'string' ? Number(value) : value;
+
+      return unformattedValue
+        .toLocaleString(navigator.language, {
+          minimumFractionDigits: decimalPrecision,
+          maximumFractionDigits: decimalPrecision,
+        })
+        .replace(',', '.');
+    }, [decimalPrecision, value]);
+
+    return (
+      <div className={classNames(styles.wrapper, className)}>
+        <div
+          className={classNames(styles.container, {
+            [styles.large]: variant === AmountInputVariant.large,
+            [styles.small]: variant === AmountInputVariant.small,
+            [styles.invalid]: invalid,
+            [styles.focus]: focused,
+            [styles.disabled]: rest.disabled,
```

Comment:
Disabled state still has background color change on `<input>`, since rest of component already has disabled behaviour/styling we can just remove the bg color styling